presto-hana/src/main/java/com/facebook/presto/plugin/hana/HanaClient.java

```diff
@@ -0,0 +1,154 @@
+import com.facebook.presto.plugin.jdbc.JdbcConnectorId;
+import com.facebook.presto.plugin.jdbc.JdbcIdentity;
+import com.facebook.presto.plugin.jdbc.JdbcTableHandle;
+import com.facebook.presto.spi.ConnectorSession;
+import com.facebook.presto.spi.ConnectorTableMetadata;
+import com.facebook.presto.spi.PrestoException;
+import com.facebook.presto.spi.SchemaTableName;
+import com.google.common.base.Joiner;
+import com.sap.db.jdbc.Driver;
+
+import javax.inject.Inject;
+
+import java.sql.Connection;
+import java.sql.DatabaseMetaData;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.Properties;
+
+import static com.facebook.presto.plugin.jdbc.DriverConnectionFactory.basicConnectionProperties;
+import static com.facebook.presto.plugin.jdbc.JdbcErrorCode.JDBC_ERROR;
+import static com.google.common.util.concurrent.MoreExecutors.directExecutor;
+import static java.lang.String.format;
+import static java.util.Locale.ENGLISH;
+
+public class HanaClient
+        extends BaseJdbcClient
+{
+    private static final Joiner DOT_JOINER = Joiner.on(".");
+
+    @Inject
+    public HanaClient(JdbcConnectorId connectorId, BaseJdbcConfig config)
+            throws SQLException
+    {
+        super(connectorId, config, "\"", connectionFactory(config));
+    }
+
+    private static ConnectionFactory connectionFactory(BaseJdbcConfig config)
+            throws SQLException
+    {
+        Properties connectionProperties = basicConnectionProperties(config);
+        return new DriverConnectionFactory(
+                new Driver(),
+                config.getConnectionUrl(),
+                Optional.ofNullable(config.getUserCredentialName()),
+                Optional.ofNullable(config.getPasswordCredentialName()),
+                connectionProperties);
+    }
+
+    @Override
+    public void abortReadConnection(Connection connection)
+            throws SQLException
+    {
+        // Abort connection before closing. Without this, the Hana driver
+        // attempts to drain the connection by reading all the results.
+        connection.abort(directExecutor());
+    }
+
+    @Override
+    public PreparedStatement getPreparedStatement(ConnectorSession session, Connection connection, String sql)
+            throws SQLException
+    {
+        PreparedStatement statement = connection.prepareStatement(sql);
+        if (statement.isWrapperFor(Statement.class)) {
+            statement.unwrap(Statement.class);
+        }
+        return statement;
+    }
+
+    @Override
+    public void createTable(ConnectorSession session, ConnectorTableMetadata tableMetadata)
+    {
+        try {
+            createTable(tableMetadata, session, tableMetadata.getTable().getTableName());
+        }
+        catch (SQLException e) {
+            throw new PrestoException(JDBC_ERROR, e);
+        }
+    }
```

Comment:
Looks like this is the same as the [base method](https://github.com/prestodb/presto/blob/7cf4582340eb0ce926a6a4560e1dcda63de443c9/presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/BaseJdbcClient.java#L314-L323) and is probably safe to remove

Reply:
Removed.